### PR TITLE
Add documentation for new trigger filters

### PR DESCRIPTION
**Pull request by devguyio**
Add documentation for new trigger filters experimental feature knative/eventing#5995

## Proposed Changes 

- Add documentation for new trigger filters experimental feature 
- Restructures the eventing experimental feature documentation

### CODE REVIEW

File: docs/eventing/experimental-features/kreference-mapping.md
Comment (odacremolbap) [inline docs/eventing/experimental-features/kreference-mapping.md:7]:
```suggestion
**Tracking issue**: [#5593](https://github.com/knative/eventing/issues/5593)
```

File: docs/eventing/experimental-features/new-trigger-filters.md
Comment (odacremolbap) [inline docs/eventing/experimental-features/new-trigger-filters.md:48]: personal preference, not an issue at all

```suggestion
* Each filter expression follows a `dialect` that defines the type of filter and the set of additional properties that are allowed within the filter expression.
```

File: docs/eventing/experimental-features/new-trigger-filters.md
Comment (odacremolbap) [inline docs/eventing/experimental-features/new-trigger-filters.md:56]:
```suggestion
CloudEvent attribute String value must match exactly the specified String value (case sensitive). 
```

File: docs/eventing/experimental-features/new-trigger-filters.md
Comment (odacremolbap) [inline docs/eventing/experimental-features/new-trigger-filters.md:121]:
```suggestion
The provided [CloudEvents SQL Expression](https://github.com/cloudevents/spec/blob/main/cesql/spec.md) must evaluate to true.
```

**Review by odacremolbap — CHANGES_REQUESTED** [review]
I think this looks very good. Might have found some small glitches (plus the `specify` spelling that reviewdog is already pointing out).

File: config/nav.yml
Comment (matzew) [inline config/nav.yml:219]: I like structuring this differently, but perhaps that stuff can be done before landing new documentation?

E.g. doing a "pre factoring" PR to do the org. stuff, and than adding the actual new doc for the trigger filters?

File: docs/eventing/experimental-features/new-trigger-filters.md
Comment (snneji) [inline docs/eventing/experimental-features/new-trigger-filters.md:157]:
```suggestion
The reason is twofold. First, at the time of developing `Trigger` APIs, there was no Subscriptions API in CloudEvents Project, so it makes sense to experiment with an API that is closer to the Subscriptions API. Second, we still want to support users workload with the old `filter` field, and give them the possibility to transition to the new `filters` field.
```

File: docs/eventing/experimental-features/new-trigger-filters.md
Comment (snneji) [inline docs/eventing/experimental-features/new-trigger-filters.md:155]:
```suggestion
### Why add yet another field? Why not make the current `filter` field more robust?
```

File: docs/eventing/experimental-features/new-trigger-filters.md
Comment (snneji) [inline docs/eventing/experimental-features/new-trigger-filters.md:159]:
```suggestion
### Why `filters` and not another name that wouldn't conflict with the `filter` field?
```

File: docs/eventing/experimental-features/new-trigger-filters.md
Comment (snneji) [inline docs/eventing/experimental-features/new-trigger-filters.md:161]:
```suggestion
We considered other names, such as `cefilters`, `subscriptionsAPIFilters`, or `enhancedFilters`, but we decided that this would be a step further from aligning with the Subscriptions API. Instead, we decided it is a good opportunity to conform with the Subscriptions API, at least at the field name level, and to leverage the safety of this being an experimental feature.
```

**Review by snneji — CHANGES_REQUESTED** [review]
Thanks for the PR. I've suggested some changes to fix some formatting errors and style.

File: config/nav.yml
Comment (devguyio) [reply config/nav.yml:219]: I don't think it's worth it TBH. It's not a huge restructure, I just split it. Feel free to suggest any other different structure.

File: docs/eventing/experimental-features/new-trigger-filters.md
Comment (devguyio) [reply docs/eventing/experimental-features/new-trigger-filters.md:25]: ack, didn't notice that one

File: docs/eventing/experimental-features/new-trigger-filters.md
Comment (devguyio) [reply docs/eventing/experimental-features/new-trigger-filters.md:62]: > Question: should `type` be indented by 2 or 4 spaces? At the moment it is 4 spaces.

`type`  should be indented with 4 spaces more than the previous `- exact:` . 

### Correct 
```yaml
spec:
  ...
  filters:
     # The value of filters is a sequence. Each sequence item is a mapping.
     # That mapping is a single key-value pair, with exact being the key.
     # And the value is another mapping, with `type` being a key and `com.github.push` is the value
    - exact:
    |___type: com.github.push  # four spaces, thus interpreted as a value of the key `exact`
```

### Wrong 
```yaml
spec:
  ...
  filters:
    - exact:
    |_type: com.github.push  # two spaces, thus interpreted as sibling key to `exact`.
```

**Comment by devguyio** [comment]
@snneji thanks a lot! That was super useful and quite some good catches!

**Comment by devguyio** [comment]
@odacremolbap comments addressed.

**Comment by odacremolbap** [comment]
All good with me, 
holding until https://github.com/knative/eventing/pull/5995 gets merged.

/hold

**Review by odacremolbap — APPROVED** [review]

**Comment by devguyio** [comment]
There's a failing e2e test
```
    --- FAIL: TestMTChannelBrokerNewTriggerFilters/New_trigger_filters/CloudEvents_SQL_filter/Assert/[Alpha/MUST_NOT]Triggers_with_new_filters_must_not_deliver_unmatched_events (0.00s)
```

**Comment by devguyio** [comment]
@odacremolbap @snneji knative/eventing#5995 is merged.

File: config/nav.yml
Comment (matzew) [reply config/nav.yml:219]: No, the structure itself is good.  i was just referencing to the fact that this PR does two things:
* restructuring
* adding doc for new trigger (as the PR title)

Could be done in two PRs. But does not need to be done in two.

File: docs/eventing/experimental-features/new-trigger-filters.md
Comment (snneji) [inline docs/eventing/experimental-features/new-trigger-filters.md:50]: needs closing tag here and for some other code snippets below

**Review by snneji — CHANGES_REQUESTED** [review]
Some of these code snippets need closing tags. Also there are a couple of minor whitespace errors. Other than that, looks good to me.

**Comment by devguyio** [comment]
@snneji applied. Thanks!

**Pull request merged**
